src/AutoRest.CSharp/Common/Input/CodeModelPartials.cs

```diff
@@ -275,189 +277,310 @@ internal partial class CodeModel
+        public ICollection<JsonPatchOp>? ResponseUpdate { get; set; }
+
+        [YamlMember(Alias = "statusCode")]
+        public int? StatusCode { get; set; }
+
+        [YamlMember(Alias = "isPrepareStep")]
+        public bool? IsPrepareStep { get; set; }
+
+        [YamlMember(Alias = "isCleanUpStep")]
+        public bool? IsCleanUpStep { get; set; }
+
+        // for TestStepArmTemplateDeployment (type==armTemplateDeployment)
+        [YamlMember(Alias = "armTemplate")]
+        public String? ArmTemplate { get; set; }
 
         [YamlMember(Alias = "armTemplatePayload")]
-        public RecordOfStringAndAny? ArmTemplatePayload;
+        public RecordOfStringAndAny? ArmTemplatePayload { get; set; }
 
-        [YamlMember(Alias = "armTemplateParametersPayload")]
-        public RecordOfStringAndAny? ArmTemplateParametersPayload;
+        [YamlMember(Alias = "armTemplatePayloadString")]
+        public string? ArmTemplatePayloadString { get; set; }
 
         // for TestStepRestCall (type==restCall)
-        [YamlMember(Alias = "operation")]
-        public RecordOfStringAndAny? Operation;
+        [YamlMember(Alias = "operationId")]
+        public String? OperationId { get; set; }
 
-        [YamlMember(Alias = "exampleId")]
-        public string? ExampleId;
+        [YamlMember(Alias = "exampleName")]
+        public string? ExampleName { get; set; }
 
-        [YamlMember(Alias = "exampleFilePath")]
-        public string? ExampleFilePath;
+        [YamlMember(Alias = "exampleFile")]
+        public string? ExampleFile { get; set; }
 
         [YamlMember(Alias = "requestParameters")]
-        public RecordOfStringAndAny? RequestParameters;
+        public RecordOfStringAndAny? RequestParameters { get; set; }
 
-        [YamlMember(Alias = "ResponseExpected")]
-        public RecordOfStringAndAny? responseExpected;
+        [YamlMember(Alias = "expectedResponse")]
+        public object? ExpectedResponse { get; set; }
 
         // test-modeler properties
         [YamlMember(Alias = "exampleModel")]
-        public ExampleModel? ExampleModel;
+        public ExampleModel? ExampleModel { get; set; }
 
-        [YamlMember(Alias = "outputVariableNames")]
-        public System.Collections.Generic.ICollection<string>? OutputVariableNames;
+        [YamlMember(Alias = "outputVariablesModel")]
+        public Dictionary<string, ICollection<OutputVariableModel>>? OutputVariablesModel { get; set; }
     };
 
-    internal partial class TestScenario
+    internal partial class OutputVariableModel
     {
-        [YamlMember(Alias = "requiredVariablesDefault")]
-        public System.Collections.Generic.Dictionary<string, string> RequiredVariablesDefault;
+        [YamlMember(Alias = "index")]
+        public int? Index { get; set; }
+
+        [YamlMember(Alias = "key")]
+        public string? Key { get; set; }
+
+        [YamlMember(Alias = "languages")]
+        public Languages? Languages { get; set; }
```

Comment:
Is this the same `Languages` object as in the code model?
In the code model, the `Languages` (they call the corresponding property `language` instead of `language`, maybe we could get some alignment here as well? if possible) is not nullable - is it possible that we could do it here as well?

Reply:
here null can specify which one in Languages/Key/Index is used in OutputVariableModel, so I think we'd better keep nullable here?